server.js

```diff
@@ -0,0 +1,50 @@
+'use strict';
+
+const http = require('http');
+const url = require('url');
+const querystring = require('querystring');
+const cowsay = require('cowsay');
+const parseBody = require('./lib/parse-body.js');
+const PORT = process.env.PORT || 3000;
+
+const server = http.createServer(function(req, res) {
+  req.url = url.parse(req.url);
+  req.url.query = querystring.parse(req.url.query);
+
+  if(req.method === 'POST') {
+    parseBody(req, function(err) {
+      if(err) return console.error(err);
+      if(req.body.text) {
+        res.writeHead(200, {'Content-Type': 'text/plain'});
+        res.write(cowsay.say({text: req.body.text}));
+      } else {
+        res.statusCode = 400;
+        res.write(400, cowsay.say({text: 'bad request'}));
+      }
+    });
+  }
+
+  if(req.method === 'GET' && req.url.pathname === '/cowsay') {
+    let params = req.url.query;
+   
+    if(params.text) {
+      res.writeHead(200, {'Content-Type': 'text/plain'});
+      res.write('hello from my server!');
+      res.write(cowsay.say({text: params.text }));
+    } else {
+      res.statusCode = 400;
+      res.write(cowsay.say({text: 'bad request'}));
+    }
+    res.end();
+  }
+
+  if(req.method === 'GET' && req.url.pathname === '/') {
+    let params = req.url.query;
+    res.write(cowsay.say(200, {text: 'Hello from my server!'}));
+    res.end();
+  }
+
+  res.end();
```

Comment:
this in some cases will cause write after end error. It can be good to either handle them all at the end or to handle them all at each path not both.